### DISCUSSION TITLE
Triton Partitioner: Any documentation for multi-constraint example?

### DISCUSSION THREAD

**Discussion by ombhilare999** [The-OpenROAD-Project/OpenROAD#4714] [General]
Hi,

I am using TritonPart in a personal project and my application closely resembled the following application mentioned in the TritonPart paper. [Here](https://vlsicad.ucsd.edu/Publications/Conferences/401/c401.pdf):

> In other words, the standard balance constraint must be satisfied
along each dimension of weights. A practical example of this
constraint is partitioning a netlist across multiple FPGAs, where
resources such as flip-flops (FFs), digital signal processing blocks
(DSPs), and look-up tables (LUTs) are all limited. This is a hard
constraint

1. In my application, I had three different types of vertices, denoted as t1, t2, and t3. My main concern was to ensure that no more than four crossings were allowed between any partitions. I believe to achieve this functionality; I need to set `-balance_constraint` to `3`.
2. In each individual partition or block, I could only have a maximum of `four t1 cells, 16 t2 cells, and four t3 cells`. This was the maximum condition. However, I was unsure about how to implement this constraint. I have read the OpenROAD paper and documentation, but any related reference would be very helpful for my application.

Thank you, let me know if you need any more information.

**Comment by bodhi91** [comment]
Please check https://github.com/The-OpenROAD-Project/OpenROAD/issues/4724
